API/Controllers/TournamentsController.cs

```diff
@@ -1,34 +1,223 @@
 using API.DTOs;
+using API.Enums;
 using API.Services.Interfaces;
+using API.Utilities;
 using Asp.Versioning;
 using Microsoft.AspNetCore.Authorization;
-using Microsoft.AspNetCore.Cors;
+using Microsoft.AspNetCore.JsonPatch;
 using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.ModelBinding;
 
 namespace API.Controllers;
 
 [ApiController]
 [ApiVersion(1)]
-[EnableCors]
 [Route("api/v{version:apiVersion}/[controller]")]
-public class TournamentsController(ITournamentsService service) : Controller
+[Authorize(Roles = "user")]
+[Authorize(Roles = "whitelist")]
+public class TournamentsController(ITournamentsService tournamentsService, IMatchesService matchesService) : Controller
 {
-    private readonly ITournamentsService _service = service;
+    private readonly ITournamentsService _tournamentsService = tournamentsService;
+    private readonly IMatchesService _matchesService = matchesService;
 
+    /// <summary>
+    /// List all tournaments
+    /// </summary>
+    /// <remarks>Will not include match data</remarks>
+    /// <response code="200">Returns all tournaments</response>
     [HttpGet]
-    [Authorize(Roles = "system")]
-    public async Task<IActionResult> GetAsync()
+    [Authorize(Roles = "admin, system")]
+    [ProducesResponseType<IEnumerable<TournamentDTO>>(StatusCodes.Status200OK)]
+    public async Task<IActionResult> ListAsync()
     {
-        IEnumerable<TournamentDTO> res = await _service.GetAllAsync();
-        return Ok(res);
+        IEnumerable<TournamentDTO> result = await _tournamentsService.ListAsync();
+        return Ok(result);
     }
 
-    [HttpPut]
-    [Authorize(Roles = "admin, system")]
-    public async Task<IActionResult> PutAsync(TournamentWebSubmissionDTO tournament)
+    /// <summary>
+    /// Submit a tournament
+    /// </summary>
+    /// <remarks>
+    /// The user submits a tournament (containing metadata and links) to the front-end. They are looking to add new data
+    /// to our database that will eventually count towards ratings.
+    ///
+    /// This post endpoint takes these links, validates them (i.e. checks for duplicates,
+    /// whether the match titles align with osu! tournament naming conventions,
+    /// amount of matches being submitted, etc.).
+    ///
+    /// Assuming we have a good batch, we will mark all of the new items as "PENDING".
```

Comment:
We mark them as pending unless `verify` is `true`. We'd then mark them as `VERIFIED` immediately.